pkg/generic/httpthrift_codec_amd64.go

```diff
@@ -0,0 +1,102 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package generic
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	athrift "github.com/apache/thrift/lib/go/thrift"
+	"github.com/cloudwego/dynamicgo/conv"
+	"github.com/cloudwego/dynamicgo/conv/j2t"
+
+	"github.com/cloudwego/kitex/pkg/generic/descriptor"
+	"github.com/cloudwego/kitex/pkg/remote"
+	"github.com/cloudwego/kitex/pkg/remote/codec/perrors"
+	cthrift "github.com/cloudwego/kitex/pkg/remote/codec/thrift"
+)
+
+func (c *httpThriftCodec) Marshal(ctx context.Context, msg remote.Message, out remote.ByteBuffer) error {
+	msgType := msg.MessageType()
+	if !c.enableDynamicgoReq || msgType == remote.Exception {
+		return c.originalMarshal(ctx, msg, out)
+	}
+
+	method := msg.RPCInfo().Invocation().MethodName()
+	if method == "" {
+		return perrors.NewProtocolErrorWithMsg("empty methodName in thrift Marshal")
+	}
+
+	data := msg.Data()
+	gArg := data.(*Args)
+	transData := gArg.Request
+	transBuff, ok := transData.(*HTTPRequest)
+	if !ok {
+		return perrors.NewProtocolErrorWithType(perrors.InvalidData, fmt.Sprintf("decode msg failed, type(%v) is not *HTTPRequest", reflect.TypeOf(transData)))
+	}
+
+	svcDsc, ok := c.svcDsc.Load().(*descriptor.ServiceDescriptor)
+	if !ok {
+		return perrors.NewProtocolErrorWithMsg("get parser ServiceDescriptor failed")
+	}
+	if svcDsc.DynamicgoDsc == nil {
+		return perrors.NewProtocolErrorWithMsg("svcDsc.DynamicgoDsc is nil")
+	}
+	tyDsc := svcDsc.DynamicgoDsc.Functions()[method].Request().Struct().Fields()[0].Type()
+
+	seqID := msg.RPCInfo().Invocation().SeqID()
+	tProt := cthrift.NewBinaryProtocol(out)
+	if err := tProt.WriteMessageBegin(method, athrift.TMessageType(msgType), seqID); err != nil {
+		return perrors.NewProtocolErrorWithMsg(fmt.Sprintf("thrift marshal, WriteMessageBegin failed: %s", err.Error()))
+	}
+	if err := tProt.WriteStructBegin(""); err != nil {
+		return perrors.NewProtocolErrorWithMsg(fmt.Sprintf("thrift marshal, WriteStructBegin failed: %s", err.Error()))
+	}
+	if err := tProt.WriteFieldBegin("", athrift.STRUCT, 1); err != nil {
+		return perrors.NewProtocolErrorWithMsg(fmt.Sprintf("thrift marshal, WriteFieldBegin failed: %s", err.Error()))
+	}
+
+	body := transBuff.GetBody()
+	buf := make([]byte, 0, len(body))
+	ctx = context.WithValue(ctx, conv.CtxKeyHTTPRequest, transBuff)
+	cv := j2t.NewBinaryConv(c.convOpts)
+	// encode with dynamicgo
+	// json []byte to thrift []byte
+	if err := cv.DoInto(ctx, tyDsc, body, &buf); err != nil {
+		return err
+	}
+
+	if _, err := out.WriteBinary(buf); err != nil {
+		return err
+	}
+	tProt.WriteFieldStop()
+	if err := tProt.WriteFieldEnd(); err != nil {
```

Comment:
as above. U can refer the code (https://github.com/cloudwego/dynamicgo/blob/main/conv/j2t/http_conv.go#L84)